desk/sur/activity.hoon

```diff
@@ -1,42 +1,74 @@
+/-  c=channels
+/+  mp=mop-extensions
 |%
-+$  stream  (mop time event)
+++  eon     ((on time event) lte)
+++  emp     ((mp time event) lte)
+++  mep     ((on time event-parent) lte)
++$  stream  ((mop time event) lte)
 +$  indices  (map index [=stream =reads])
-+$  reads  [floor=time (set event-id=time)]
-+$  derived-unreads  [time count=@ud threads=(list [time count=@ud])]
++$  reads
+  $:  floor=time      ::  latest time above which everything is read
+      =event-parents  ::
+  ==
++$  event-parent  [seen=? reply-floor=time]
++$  event-parents  ((mop timid event-parent) lte)
 +$  index
   $%  [%channel channel-concern]
       [%dm dm-concern]
   ==
-+$  group-concern    group=flag
-+$  channel-concern  [channel=nest group=flag]
-+$  dm-concern       =whom
-+$  post-concern     [=message-key channel=nest group=flag]
-+$  reply-concern    [=message-key target=message-key channel=nest group=flag]
-+$  dm-post-concern  [=writ-key =whom]
-::  $event: an instance of activity
-::
-::    $flavor: what activity generated the event and where did it happen
-::    $content: text of the message that generated the event or a description
-::    $level: what "level" of importance this is deemed by originator
-::    $read: has this event been seen or interacted with
-::
-+$  event  [=flavor =level]
-+$  level  ?(%notify %default %trivial)
-+$  content
-  ::  same as content of actual message
-  ~
++$  volume  (map flavor level)
```

Comment:
I'm not sure we want volume per type, instead by source like it currently is in groups.